compiler/rustc_codegen_gcc/src/lib.rs

```diff
@@ -35,7 +35,7 @@ extern crate rustc_middle;
 extern crate rustc_session;
 extern crate rustc_span;
 extern crate rustc_target;
-extern crate tempfile;
+
```

Comment:
`tempfile` is used at [line 89](https://github.com/rust-lang/rust/pull/112370/files#diff-e4d3a952d65504b0ae63b6c1da8c89b2b56cbdbbfda77c6e46e1d4f348dbd09fL89) so it should not be removed.

Not sure why there's a warning emitted about this, but removing this line will result in a compilation error.